services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/crt/S3MetaRequestPauseObservable.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.services.s3.internal.crt;
+
+import java.util.function.Function;
+import software.amazon.awssdk.annotations.SdkInternalApi;
+import software.amazon.awssdk.crt.s3.S3MetaRequest;
+
+@SdkInternalApi
+public class S3MetaRequestPauseObservable {
```

Comment:
The naming might be confusing. LMK if you can think of a better name, but this is the class created in Transfer Manager and passed to `S3CrtAsyncHttpClient` through `SdkHttpExecutionAttribute`.

I thought about using generic types instead of String, but it's a bit awkward in  `SdkHttpExecutionAttribute` because we'd need to do casting for generic types

```
    public static final S3InternalSdkHttpExecutionAttribute<S3MetaRequestPauseObservable> METAREQUEST_PAUSE_OBSERVABLE =
        new S3InternalSdkHttpExecutionAttribute<>(S3MetaRequestPauseObservable.class);
```

Reply:
Could you please add a Java Doc for this class to know the intent/responsibility of this task , just for internal references.